### PR TITLE
fix(rome_js_analyze): suppress noUnusedVariables's false positive diagnostics when using indexed type

### LINKED ISSUE #4634: 🐛 correctness/noUnusedVariables false positive when using indexed type
### Environment information

```block
playground
```

### What happened?

https://docs.rome.tools/playground/?lintRules=all&code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABmAGEAaQBsAHMAKABuAGEAbQBlADoAIABzAHQAcgBpAG4AZwApACAAewAKACAAIAByAGUAdAB1AHIAbgAgAHsACgAgACAAIAAgAG4AYQBtAGUALAAKACAAIAB9ACAAYQBzACAAewAgAFsAbgBhAG0AZQA6ACAAcwB0AHIAaQBuAGcAXQA6ACAAcwB0AHIAaQBuAGcAIAB9ADsACgB9AAoACgBlAHgAcABvAHIAdAAgAGYAdQBuAGMAdABpAG8AbgAgAHAAYQBzAHMAKABuAGEAbQBlADoAIABzAHQAcgBpAG4AZwApACAAewAKACAAIAByAGUAdAB1AHIAbgAgAHsACgAgACAAIAAgAG4AYQBtAGUALAAKACAAIAB9ACAAYQBzACAAewAgAFsAbgBvAHQATgBhAG0AZQA6ACAAcwB0AHIAaQBuAGcAXQA6ACAAcwB0AHIAaQBuAGcAIAB9ADsACgB9AAoACgAKAAoA

image

### Expected result

should not error

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

### PR DESCRIPTION
**Pull request by nissy-dev**
## Summary

Fix #4634 

## Test Plan

I update some snapshot tests.

### CODE REVIEW

File: crates/rome_js_analyze/tests/specs/suspicious/noRedeclare/invalid.jsonc [hunk 1]
```diff
@@ -25,6 +25,5 @@
 	"function f() { var a; if (test) { var a; } }",
 	"for (var a, a;;);",
 	"for (;;){ var a, a,;}",
-	"function f(x) { var x = 5; }",
-  "interface A { [index: number]: string; [index: number]: string; }"
```
Comment (nissy-dev) [inline crates/rome_js_analyze/tests/specs/suspicious/noRedeclare/invalid.jsonc:29]: Semantic analyzer no longer handle identifiers of type variable, so I removed.

In this case, tsc throw the syntax error.
Thus, I seem that it is better to make this rule in syntax rule like noDuplicatePrivateClassMembers.

https://typescript-eslint.io/play/#ts=4.9.3&jsx=true&sourceType=module&code=JYOwLgpgTgZghgYwgAgILIN7IFDOQbVABMIAPALmQGcwpQBzAXUprpHoG5cDiyXaGzagPZcAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

File: crates/rome_js_analyze/src/semantic_analyzers/suspicious/no_redeclare.rs [hunk 2]
```diff
@@ -128,72 +126,19 @@ fn check_redeclarations_in_single_scope(scope: &Scope, redeclarations: &mut Vec<
...
-                            name,
-                            declaration: *first_text_range,
-                            redeclaration: id_binding.syntax().text_trimmed_range(),
-                        }),
-                    }
+                    redeclarations.push(Redeclaration {
+                        name,
+                        declaration: *first_text_range,
+                        redeclaration: id_binding.syntax().text_trimmed_range(),
+                    })
```
Comment (nissy-dev) [inline crates/rome_js_analyze/src/semantic_analyzers/suspicious/no_redeclare.rs:137]: This rule no longer handles the case like `interface A { [index: number]: string; [index: number]: string; }`, so I removed the unnecessary logic.

**Review by Conaclos — APPROVED** [review]

File: xtask/codegen/js.ungram [hunk 3]
```diff
@@ -2066,9 +2066,11 @@ TsIndexSignatureTypeMember =
 // { [a: string]: number }
 //    ^^^^^^^^^
 TsIndexSignatureParameter =
-	binding: JsIdentifierBinding
+	binding: TsIndexSignatureParameterName
```
Comment (nissy-dev) [inline xtask/codegen/js.ungram:2069]: I introduce the new CST node for the binding of TsIndexSignatureParameter. I think this binding is not JsIdentifierBinding.

**Comment by nissy-dev** [comment]
This PR doesn't pass some tests, so I will rework and create the another PR.

**Pull request closed without merging**